docs/components/modeler/web-modeler/milestones.md

```diff
@@ -1,50 +1,158 @@
 
 :::note
 Milestones of resources belonging to a process application are tied to the main process and cannot be modified.
 :::
 
-### Restoring milestones
+## Compare milestones
+
+You can compare the change history between two milestones, either visually as a diagram or as code in an XML diff layout.
+
+1. Open the milestone history for your diagram.
+1. Ensure that the **Show changes** toggle in the **Changes** panel is turned on.
+1. Select the milestone that you want to compare. The previous milestone is automatically selected for comparison.
+
+:::note
+
+Turn off the **Show changes** toggle if you want to view individual milestones without comparison to the previous milestone.
+
+:::
+
+### Compare milestones in visual view
+
+To view BPMN diagram changes visually, select the **Visual view** tab.
+
+![milestones diffing](img/milestones/web-modeler-milestone-visual-diffing.png)
+
+- Differences between the milestones are highlighted visually on the diagram. For example, if an element was added, this change is highlighted in green with a plus symbol. Hover over a change to view more details.
+- Only differences that affect the execution of the BPMN process are highlighted.
+- The **Changes** panel shows the details of each change, including the type and identifier. Select a change to highlight it.
+
+:::note
+
+You can only use the **Code view** to compare changes in a DMN diagram. The **Visual view** only shows a view of the milestone.
+
+:::
+
+### Compare milestones in code view
 
-Hover over a milestone, click on the three vertical dots, and expand for more options.
-![milestones restore](img/milestones/web-modeler-milestone-restore.png)
-![milestones restore](img/milestones/web-modeler-milestone-restore-complete.png)
+To view BPMN and DMN diagram changes as code in an XML diff layout, select the **Code view** tab.
 
-### Comparing milestones
+![milestones diffing](img/milestones/web-modeler-milestone-code-diffing.png)
 
-Milestones can be compared visually. By enabling the diffing feature, the currently selected milestone is compared to its predecessor.
+- The XML for the previous milestone is shown on the left, with the currently selected milestone shown on the right.
+- Differences between the milestones are highlighted in the XML. For example, if an element was added, this change is highlighted in green.
+- Only differences that affect the execution of the BPMN process are highlighted.
 
-The differences that are highlighted are only those that affect the execution of the BPMN process. Pure visual changes like position changes are not highlighted.
-![milestones diffing](img/milestones/web-modeler-milestone-diffing.png)
+### Compare process application milestones
+
+You can also use the **Code view** tab to view and compare the [versioned milestones](process-applications.md#versioning) of individual files in a process application.
```

Comment:
❌ It is wrong to say "You can use the Code view tab to view and compare milestones of individual files"

For process applications, when a user opens the milestone history for any file, the history page includes a file switcher which can be used to switch between all files in the process application to see their individual milestones.

❌ The title `Compare process application milestones` also does not make sense.

There is no concept of process application milestone. Within process applications, there is the concept of bulk creation of milestones. If a user tries to create a milestone for an individual resource, milestones are created for all resources present in the process application at that time.

❌ I also think this whole section makes more sense to be included in the process applications page under versioning to avoid confusion. There is some limitation wrt process applications milestones creation which can make it confusing. 
- Users might expect versioning of the process application (this does not exist currently)
- What exists is bulk creation of milestones so that users don't have to go to individual files and create milestones separately to save a snapshot.

Check out the [kick-off document](https://docs.google.com/document/d/1jQvjZQkU1mi5WKkb6yq0Cty4fXLlfozynh4j1tK2hto/edit#heading=h.lntg56ljm653) to get a better understanding of the limitation I'm talking about. We need to make this limitation clear through the documentation.

Reply:
Thanks, I'll review this whole section 👍

Reply:
I find it really nice how you revised it ❤️

Reply:
@nithinssabu I've modified the versioning section following your comments - please review 👍